test/unit/ProgamCallUnit.js

```diff
@@ -174,11 +182,13 @@ describe('ProgramCall Class Unit Tests', () => {
       const pgm = new ProgramCall('MYPGM', { lib: 'MYLIB', func: 'MY_PROCEDURE' });
 
       const params = [
-        [0, '3s0'],
-        [0, '7s0', { name: 'ds_fld2' }],
+        { type: '3s0', value: 0 },
+        { type: '7s0', value: 0, name: 'ds_fld2' },
       ];
 
-      pgm.addParam(params, { name: 'inds', by: 'val', io: 'both' });
+      pgm.addParam({
+        fields: params, type: 'ds', name: 'inds', by: 'val', io: 'both',
```

Comment:
Prefer to see attribute order as name, then type, then the rest. fields probably makes most sense to go last